src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.ObsoletedInOSPlatformTests.cs

```diff
@@ -0,0 +1,547 @@
+    }
+    
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"", Url = ""http://www/look.for.more.info"")]
+    public void ObsoletedWithUrl() { }
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"", ""Use other method instead"", Url = ""http://www/look.for.more.info"")]
+    public void ObsoletedWithMessageAndUrl() { }
+    [ObsoletedInOSPlatform(""Windows10.1.1.1"", ""Use other method instead"")]
+    public void ObsoletedWithMessage() { }
+}" + MockObsoletedAttributeCS;
+            await VerifyAnalyzerCSAsync(csSource, s_msBuildPlatforms, VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsAllPlatforms).WithLocation(0).
+                    WithArguments("Test.ObsoletedWithMessageAndUrl()", "'Windows' 10.1.1.1 and later (Use other method instead http://www/look.for.more.info)"),
+                  VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsAllPlatforms).WithLocation(1).
+                    WithArguments("Test.ObsoletedWithMessage()", "'Windows' 10.1.1.1 and later (Use other method instead)"),
+                  VerifyCS.Diagnostic(PlatformCompatibilityAnalyzer.ObsoletedCsAllPlatforms).WithLocation(2).
+                    WithArguments("Test.ObsoletedWithUrl()", "'Windows' 10.1.1.1 and later (http://www/look.for.more.info)"));
+        }
+
+        [Fact]
+        public async Task ObsoletedAPIsCAlledFromDifferentCallsite()
+        {
+            var csSource = @"
+using System;
+using System.Runtime.Versioning;
+using Mock;
+
+public class Test
+{
+    [SupportedOSPlatform(""IOS"")]
+    public void CallsiteReachableOnIOS()
+    {
+        ObsoletedOnWindows(); // Unreachable on windows, not warn
+        ObsoletedWithMessage();
+        {|CA1422:ObsoletedOnIOS14()|}; // This call site is reachable on: 'IOS', 'maccatalyst'. 'Test.ObsoletedOnIOS14()' is obsoleted on: 'ios' 14.0 and later, 'maccatalyst' 14.0 and later.
+    }
+
+    [Mock.UnsupportedOSPlatform(""ios13.0"")]
+    [System.Runtime.Versioning.UnsupportedOSPlatform(""Windows10.1.0"")]
+    public void SuppressedByCallsiteUnsupported()
+    {
+        ObsoletedOnWindows(); // Not supported on windows with matching version, not warn
+        ObsoletedWithMessage(); // Same as above
+        ObsoletedOnIOS14();
+    }
+
+    [Mock.UnsupportedOSPlatform(""ios15.0"")]
+    [System.Runtime.Versioning.UnsupportedOSPlatform(""Windows11.1.0"")]
+    public void NotSuppressedByCallsiteUnsupported() // obsoleted before unsupported version
+    {
+        {|CA1422:ObsoletedOnWindows()|}; // This call site is reachable on: 'Windows' 11.1.0 and before. 'Test.ObsoletedOnWindows()' is obsoleted on: 'Windows' 10.1.1.1 and later.
+        {|CA1422:ObsoletedWithMessage()|}; // This call site is reachable on: 'Windows' 11.1.0 and before. 'Test.ObsoletedWithMessage()' is obsoleted on: 'Windows' 10.1.1.1 and later, Use other method instead.
```

Comment:
Nit, I think this comment is out of date.
```suggestion
        {|CA1422:ObsoletedWithMessage()|}; // This call site is reachable on: 'Windows' 11.1.0 and before. 'Test.ObsoletedWithMessage()' is obsoleted on: 'Windows' 10.1.1.1 and later (Use other method instead).
```